tests/Console/E2ETest.php

```diff
@@ -0,0 +1,230 @@
+
+    protected function setUp()
+    {
+        // E2E tests usually require to chdir to their location
+        // Hence we would need to go back to this dir
+        $this->cwd = \getcwd();
+    }
+
+    /**
+     * Longest test: runs under about 160-200 sec
+     *
+     * To be run with:
+     *
+     * php -dmemory_limit=128M vendor/bin/phpunit --group=large
+     *
+     * @group e2e
+     * @large
+     */
+    public function test_it_runs_on_itself()
+    {
+        $this->requiresMemoryLimitSet();
+
+        $output = $this->runInfection(self::EXPECT_SUCCESS, [
+            '--test-framework-options="--exclude-group=e2e"',
+        ]);
+
+        $this->assertRegExp('/\d+ mutations were generated/', $output);
+        $this->assertRegExp('/\d{2,} mutants were killed/', $output);
+    }
+
+    /**
+     * @group e2e
+     */
+    public function test_it_runs_configure_command_if_no_configuration()
+    {
+        chdir('tests/Fixtures/e2e/Unconfigured/');
+
+        $output = $this->runInfection(self::EXPECT_ERROR);
+
+        $this->assertContains(ConfigureCommand::NONINTERACTIVE_MODE_ERROR, $output);
+    }
+
+    /**
+     * @dataProvider e2eTestSuiteDataProvider
+     * @group e2e
+     */
+    public function test_it_runs_an_e2e_test_with_success(string $dirName)
+    {
+        $this->runOnE2EFixture($dirName);
+    }
+
+    public function e2eTestSuiteDataProvider(): \Generator
+    {
+        $directories = Finder::create()
+            ->depth('== 0')
+            ->in(__DIR__ . '/../Fixtures/e2e/')
+            ->directories();
+
+        foreach ($directories as $dirName) {
+            if (file_exists($dirName . '/run_tests.bash')) {
+                // skipping non-standard tests
+                // specifically Memory_Limit - it is very slow to fail
+                continue;
+            }
+
+            yield [
+                $dirName,
+            ];
+        }
+    }
+
+    protected function tearDown()
+    {
+        if ($this->previousLoader) {
+            $this->previousLoader->unregister();
+        }
+
+        \chdir($this->cwd);
+    }
+
+    private function runOnE2EFixture($path): string
+    {
+        $this->assertTrue(is_dir($path));
+        chdir($path);
+
+        $this->installComposerDeps();
+        $output = $this->runInfection(self::EXPECT_SUCCESS);
+
+        $this->assertRegExp('/You are running Infection with \w+ enabled./', $output);
+        $this->assertRegExp('/\d+ mutations were generated/', $output);
+        $this->assertRegExp('/\d+ mutants were killed/', $output);
+
+        if (isset($_SERVER['GOLDEN'])) {
+            copy('infection-log.txt', 'expected-output.txt');
+            $this->markTestSkipped('Saved golden output');
+        }
+
+        $this->assertFileEquals('expected-output.txt', 'infection-log.txt', sprintf('%s/expected-output.txt is not same as infection-log.txt', getcwd()));
+
+        return $output;
+    }
+
+    private function installComposerDeps()
```

Comment:
So if I understand correctly, here we install the deps + autoload the classes for the e2e test to make sure that if there's an auto-loading issue it's breaking here and not elsewhere because it would be harder to debug?

Reply:
That’s right

Reply:
Well, that can be a problem. I can `include` all classes right away [instead of feeding them to the autoloader here](https://github.com/infection/infection/commit/287ec5e2e063643880e00b45d7828d2fbf5f15b6#diff-c50be878dbd21bc70fc3c4da7deedccfR144), but E2E tests could have (and they have I think) absolutely same classes, with the same content, and the same name. That's going to be an error. 

Infection sub-processes use actual files, but if it some point they happen to be different, this will be seen in different finals, and test will fail in this case.